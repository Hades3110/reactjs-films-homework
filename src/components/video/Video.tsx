import { useEffect, useState } from "react"
import ReactPlayer from "react-player"
import { useDispatch, useSelector } from "react-redux"
import { ActionTypeInterface } from "../../global"
import { changeVideoWindow } from "../../redux/filmVideoPlay/action"
import { getFilmVideo } from "../../services/filmService"
import styles from './video.module.scss'

const Video = () => {

    const [key, setKey] = useState<string>('')

    const dispatch = useDispatch()
    const watch = useSelector((state: ActionTypeInterface) => state.videoPlayer.watch)
    const id = useSelector((state: ActionTypeInterface) => state.videoPlayer.id)

    useEffect(() => {
        if (id) getFilmVideo(id).then(res => setKey(res.results[0].key))
    })
    console.log(key)

    return (<>
        {watch ? <div className={styles.video} onClick={() => {
            dispatch(changeVideoWindow(false, 0))
        }
        }>
            {key ? <><ReactPlayer
                className={styles.player}
                url={`https://www.youtube.com/watch?v=${key}`}
                controls={true}
                width={'60%'}
                height={'70vh'}
            /> <div className={styles.close}>&#x2715;</div>
            </> : <div className={styles.notFound}><div>Video is not found</div><button>Close</button></div>}
        </div> : ''}
    </>)
}

export default Video