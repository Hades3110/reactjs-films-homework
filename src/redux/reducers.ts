import { combineReducers } from "redux";
import filmListPageCountReducer from "./filmListCountAndCategories/reducer";
import filmListSortReducer from "./filmListSort/reducer";
import mainBannerReducer from "./mainBanner/reducer";

const rootReducer = combineReducers({
    pageCounter: filmListPageCountReducer,
    sortType: filmListSortReducer,
    mainBanner: mainBannerReducer
})

export default rootReducer
