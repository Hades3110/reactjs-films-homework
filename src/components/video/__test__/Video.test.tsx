import { create } from "react-test-renderer";
import Video from '../Video';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../../../redux/reducers';

describe('Video Component', () => {

  const store = createStore(rootReducer)

  it('Video snapshot test', () => {
    const component = create(<Provider store={store}><Video /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});