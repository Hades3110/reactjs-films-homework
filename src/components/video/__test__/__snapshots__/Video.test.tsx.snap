// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Component Video snapshot test 1`] = `""`;
